### PR TITLE
Support fetchRows, head, etc. for DDF with complex type columns 

### PR DESCRIPTION
**Pull request by namma**
sql2txt implementation has been updated. Value of a non-primitive column is now represented by JSON.

https://adatao.atlassian.net/browse/PE-514

### CODE REVIEW

**Comment by khangich** [comment]
ok to test

File: spark/src/main/java/io/spark/ddf/etl/SqlHandler.java [hunk 1]
```diff
@@ -109,12 +115,19 @@ public DDF sql2ddf(String command, Schema schema, String dataSource, DataFormat
...
     List<String> lsString = new ArrayList<String>();
 
     for (Row row : arrRow) {
       lsString.add(row.mkString("\t"));
     }
-    return lsString;
+    return lsString
+    */
+
+    String[] strResult = SparkUtils.jsonForComplexType(rdd, "\t");
```
Comment (Huandao0812) [inline spark/src/main/java/io/spark/ddf/etl/SqlHandler.java]: use DataFrame.toJson to convert the DataFrame to RDD[String]
https://github.com/apache/spark/blob/branch-1.3/sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala#L1236

@namma

File: spark/src/main/scala/io/spark/ddf/util/SparkUtils.scala [hunk 2]
```diff
@@ -119,6 +124,74 @@ object SparkUtils {
...
+   * @param df
+   * @return an Array of string showing the dataframe with complex column-object replaced by json string
+   */
+  def jsonForComplexType(df: DataFrame, sep: String): Array[String] = {
+    val schema = df.schema
+    val df1: RDD[String] = df.map(r => rowToJSON(schema, r, sep))
+    df1.collect()
+  }
+
+  private def rowToJSON(rowSchema: StructType, row: Row, separator: String): String = {
```
Comment (Huandao0812) [inline spark/src/main/scala/io/spark/ddf/util/SparkUtils.scala]: don't need to do this, use https://github.com/apache/spark/blob/branch-1.3/sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala#L1236

File: spark/src/main/java/io/spark/ddf/etl/SqlHandler.java (hunk 1 above)
Comment (namma) [inline spark/src/main/java/io/spark/ddf/etl/SqlHandler.java]: no, it is different. It is to convert entire row to JSON. We only convert each column object to JSON.

File: spark/src/main/scala/io/spark/ddf/util/SparkUtils.scala (hunk 2 above)
Comment (namma) [inline spark/src/main/scala/io/spark/ddf/util/SparkUtils.scala]: are you sure ;)

**Pull request merged**